src/imitation/policies/trainer.py

```diff
@@ -0,0 +1,83 @@
+"""Training policies with a specifiable reward function and collect trajectories."""
+from typing import List, Union
+
+from stable_baselines3.common.base_class import BaseAlgorithm
+from stable_baselines3.common.vec_env import VecEnv
+
+from imitation.data import types
+from imitation.data.wrappers import BufferingWrapper
+from imitation.rewards.common import RewardFn
+from imitation.rewards.reward_nets import RewardNet
+from imitation.util.reward_wrapper import RewardVecEnvWrapper
+
+
+class AgentTrainer:
```

Comment:
Do you expect `AgentTrainer` to be used in other algorithms in `imitation`? I don't object it to being in its own file, just trying to understand why you split this up, and not e.g. the fragmenter, gatherer, etc.

Reply:
Yes, I thought `AgentTrainer` might be useful elsewhere (not even necessarily by other algorithms, but maybe by end users instead). I basically just think of it as a utility wrapper for training an SB3 algorithm on a reward function different than the one provided by the environment.

Admittedly, as long as we don't want to collect trajectories during training, it's probably simpler to just use the `RewardVecEnvWrapper` manually. And I might end up adding a bit more code to `AgentTrainer` that would rarely be useful outside preference comparisons. I'll leave this open for now and decide later whether the final version makes sense standing on its own